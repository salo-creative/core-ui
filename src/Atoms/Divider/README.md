# Divider

Divider is a pure styled component render. Renders a standard `<hr>` with themed styling

---

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import Divider from '@salo/core-ui/Atoms/Divider';
```

Implement as follows

```javascript
<Divider
  height={ 2 }
  color='#00aced'
  margin='2rem 0'
>
```

**For full prop types and usage see storybook info/knobs**