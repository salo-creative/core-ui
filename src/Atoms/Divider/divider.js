import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// HELPERS
import { colours } from '../../helpers/colours';

const StyledDivider = styled.hr`
  border: 0;
  background: ${ ({ color }) => color };
  height: ${ ({ height }) => `${ height }px` };
  margin: ${ ({ margin }) => margin };
`;

const Divider = (props) => {
  return <StyledDivider { ...props } />;
};

Divider.defaultProps = {
  color: colours.grey,
  height: 1,
  margin: '2rem 0'
};

Divider.propTypes = {
  color: PropTypes.string,
  height: PropTypes.number,
  margin: PropTypes.string
};

export default Divider;