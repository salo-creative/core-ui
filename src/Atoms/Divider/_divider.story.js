import React from 'react';

// load tests
import { withTests } from '@storybook/addon-jest';
import {
  withKnobs, text, number, color
} from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { Divider, colours } from '../../index';

// README //
import README from './README.md';

// Start of story logic
export const Basic = () => {
  const height = number('Height', 1);
  const colour = color('Colour', colours.grey);
  const margin = text('Margin', '2rem 0');
  return <Divider height={ height } color={ colour } margin={ margin } />;
};

Basic.story = {
  decorators: [withKnobs, withTests({
    results
  })],
  parameters: {
    notes: README,
    jest: ['divider']
  }
};

export default {
  title: 'Atoms/Divider'
};