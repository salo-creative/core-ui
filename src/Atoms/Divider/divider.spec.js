import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import Divider from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <Divider />
  );
  expect(container.firstChild).toMatchSnapshot();
});