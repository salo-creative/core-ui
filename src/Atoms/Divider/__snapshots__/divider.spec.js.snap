// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c0 {
  border: 0;
  background: #e9ecef;
  height: 1px;
  margin: 2rem 0;
}

<hr
  class="c0"
  color="#e9ecef"
  height="1"
/>
`;
